File: P1.3/P1.3_opcional/src/FilterChain.dart
import 'Filter.dart';
import 'Target.dart';

class FilterChain {
  late List<Filter> _filters;

  FilterChain() {
    _filters = [];
  }

  addFilter(Filter f) => _filters.add(f);

  execute(Target t) {
    _filters.forEach((element) {
      element.execute(t);
    });
  }
}

File: P1.3/P1.3_obligatorio/src/FilterChain.cpp
#include "FilterChain.h"
#include <iostream>
FilterChain::FilterChain() {
    this->filters = std::list<Filter*>();
}

void FilterChain::addFilter(Filter* & f) {
    this->filters.push_back(f);
}

void FilterChain::execute(Target & t) {
    for(Filter* f : filters) {
        f->execute(t);
    }
}
